homeassistant/components/binary_sensor/zha.py

```diff
@@ -87,3 +87,31 @@ def cluster_command(self, aps_frame, tsn, command_id, args):
         elif command_id == 1:
             _LOGGER.debug("Enroll requested")
             self.hass.add_job(self._ias_zone_cluster.enroll_response(0, 0))
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        _LOGGER.debug("%s async_update", self.entity_id)
+
+        @asyncio.coroutine
+        def safe_read(cluster, attributes):
```

Comment:
Please extract this method and move it to the `zha` component and import it from there. There is no need to copy this method to every platform.

Reply:
I may be wrong here but I think `should_poll(self) -> bool:` also needs to be overridden to return false on the base ZHA entity class. With debugging on, it appears that hass is calling the `async_update` functions that were added to these classes to poll for changes behind the scenes. The zigbee devices are already configured to push changes to hass when necessary to preserve battery life. I think leaving this as is will severly affect battery life of the sensors. @balloob, does this make sense? or am I way off base?